homework-3/student_submission.cpp

```diff
@@ -65,20 +71,30 @@ ProblemQueue problemQueue;
 // generate numProblems sha1 hashes with leadingZerosProblem leading zero bits
 // This method is intentionally compute intense so you can already start working on solving
 // problems while more problems are generated
-void generateProblem(int seed, int numProblems, int leadingZerosProblem){
-    generating = true;
-    srand(seed+1);
+void generateProblem(std::string problemRand[], int numProblems, int leadingZerosProblem, int idx){
+    // std::cout << numProblems << "\n";
+    Sha1Hash mock;
 
-    for(int i = 0; i < numProblems; i++){
-        std::string base = std::to_string(rand()) + std::to_string(rand());
-        Sha1Hash hash = Utility::sha1(base);
+    int step = numProblems / NUM_GENERATORS;
+    if (! (numProblems % NUM_GENERATORS == 0)) {std::cout << "ass error";exit(1);}
+
+    for(int i = idx*step; i < (idx+1)*step; i++){
```

Comment:
Create a thread pool instead of splitting the work here

Reply:
That's not needed, the worker threads never extinguish the thread pool while the generator are working as we have way more workers than generators.

Reply:
>  the worker threads never extinguish the thread pool 

Please clarify this sentence


> we have way more workers than generators

What if I want to change the number of generators and workers to try how it affects performance

Reply:
> What if I want to change the number of generators and workers to try how it affects performance

That's out of the scope of the project.